### PR TITLE
Add Rudamentary Intake

### LINKED ISSUE #46: Make an Intake
**Purpose**
The purpose of this addition is to give our robot the ability to take in notes.

**Project Scope**
- [ ] Add an intake subsystem
- [ ] Simulate it (within reason)
- [ ] Establish an API

### PR DESCRIPTION
**Pull request by charlesmackenzie**
I've written a basic intake subsystem and some code to puppeteer a simulated note through the intake/indexer system. 

In the future, I would prefer to use a timer of some kind to simulate the movement of a note

Closes #46

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                break;
+            case SEEKING:
+                seeking();
+                break;
+            case FEEDING:
+                feeding();
+                break;
+            case PASSING:
+                passing();
+                break;
```
Comment (jkleiber) [inline src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java:39]: can we add the state transition logic to our google sheet on state machines

**Review by jkleiber — COMMENTED** [review]
one comment before I dive into the state machine logic

**Comment by charlesmackenzie** [comment]
CI is passing now, but I still need to add a 'reversing' state and make the diagram. I'll re-request a review then.

File: src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        }
+    }
+
+    private void seeking() {
+        if (inputs.noteSensed) {
+            state = State.PASSING;
+        }
+
+        if (action == IntakeAction.REVERSE) {
+            io.setBeltVoltage(-5);
```
Comment (jkleiber) [inline src/main/java/frc/robot/subsystems/intake/IntakeSubsystem.java:86]: We should make this (and other places in the code) into constants / configurable values on the dashboard

**Review by jkleiber — APPROVED** [review]
approved with a non blocking comment

**Review by redPlover — APPROVED** [review]
Looks good

**Review by jkleiber — APPROVED** [review]

**Pull request merged**